crates/symbolicator/src/services/download/gcs.rs

```diff
@@ -205,6 +205,14 @@ impl GcsDownloader {
         Ok(token)
     }
 
+    /// Downloads a source hosted on GCS.
+    ///
+    /// # Errors
+    /// - [`GcsError::InvalidUrl`]: the url was malformed and/or could not be parsed
+    /// - [`DownloadError::Reqwest`]: unable to begin streaming the source, or the initial HEAD
+    ///   request encountered an error
+    /// - [`DownloadError::Rejected`]: the initial HEAD request received a non-successful response
+    /// - [`DownloadError::Cancelled`]: the request timed out
```

Comment:
in general these descriptions should probably be documented on the error types instead?  Not that i mind too much, but having this in every use might become tricky to all keep up to date.

Reply:
Good point; I added all of these in because I started finding it a bit tedious constantly comparing the `download_source` implementations across all sources and seeing which errors were actually thrown by them while I was updating its consumers (`dispatch_download`, `download`, data cache itself, etc.)

I'll double check to see if all of the download error variants are consistently being used in the same spots, and move them if it doesn't differ from one source's `download_source` to another (e.g. does HTTP return `DownloadError::Request` for other things? does GCS? does S3?)